src/hooks/useUser.js

```diff
@@ -0,0 +1,36 @@
+import { get } from 'lodash'
+import { isEnabled, useAuth as useOidcAuth } from 'lib/auth/oidc'
+import { useAuth as useForemanAuth } from 'lib/auth/foreman'
+
+export function useForemanUser() {
+  const { userData, signOut } = useForemanAuth();
+
+  return userData ? {
+    username: get(userData, 'user.fullname', null),
+    token: get(userData, 'token', null),
+    signOut
+  } : null
+}
+
+export function useOidcUser() {
+  const { userData, signOut } = useOidcAuth();
+
+  if(!isEnabled) {
+    return null
+  }
+
+  return userData ? {
+    username: get(userData, 'profile.preferred_username', null),
```

Comment:
the `profile.preferred_username` field should also be configurable with a setting. Sadly not all IdP's follow the same scheme or are configured the same way.